topics/instructors/tutorials/galaxy-admin-training/tutorial.md

```diff
@@ -0,0 +1,228 @@
+- Updates packages (slow)
+- Installs basic dependencies (emacs/vim/nano, git, etc.)
+- Adds `gat-cli` script to `/usr/bin/gat`
+- Optionally sets a password to the machine
+	- If you set `set_password=true` in the hosts file, you can set a password on machines.
+	- This is useful when your machines only have an SSH key on them, and no password set for students.
+- Reboots the machines
+
+# Testing
+
+Once your VMs are ready you should test ALL of the lessons you intend to teach. Most of the training should be fine, but sometimes changes in e.g. Galaxy versions or the availability of newer versions of the ansible modules will indicate you should test the training as you plan to teach it, and update the training materials where relevant.
+
+# Starting Your Training
+
+We recommend providing a website [similar to our GitHub repository](https://github.com/galaxyproject/admin-training/tree/2020-barcelona) (or using our repository! Ask us for a branch.) with at minimum the following links:
+
+- **Q&A** pointing to a Google document where students ask their questions.
+	- This is an excellent format for discussion as it allows students to ask as many questions and responses can be "threaded" by default.
+	- Additionally images and complex formatting is easy
+	- Lastly, it's *anonymous* which many students prefer.
+- **Chat** pointing to `https://gxy.io/gatchat` (or your preferred channel)
+- **VM List** pointing to `https://gxy.io/gatmachines` (or your own spreadsheet)
+- The slides and tutorials for your training.
+
+# During Your Training
+
+During planning for BCC2020 we found that monitoring student progress would be extremely difficult, it was not a situation we had encountered before as this was first remote GAT during the pandemic. So we developed a small utility, the `gat-cli`, which assists in monitoring students' progress.
+
+```
+$ gat
+Galaxy Admin Training (gat) tool:
+
+  gat status-ansible     [Admin] Check status of ansible training
+  gat status-galaxy      [Admin] Check status of ansible-galaxy training
+  gat status-cvmfs       [Admin] Check status of cvmfs training
+  gat status-pulsar      [Admin] Check status of pulsar training
+```
+
+Each of these commands run a couple checks on their local machine. For example the `status-galaxy` command checks:
+
+- Check that the `postgresql` service is running
+- Does a db exists named `galaxy`
+- Does http://localhost:8080/api/version respond with *some* content
+- Check that the `galaxy` service is running
+- Check that the `nginx` service is running
+- Does https://localhost:443/api/version respond with *some* content
+
+However, the `gat` command is only available on the VMs, so we've written an ansible task (wrapped in the Makefile) which SSHs into every machine, and runs the gat command for you:
+
+```
+$ make check-galaxy-eu
```

Comment:
I guess this will work only if the VMs are hosted by UseGalaxy.eu ? Not sure how complicated it is to make it usable for non-EU GATs.

Reply:
As long as we get fundet we are happy to provide those resource for all kind of GATs.

Reply:
Sure, but there could be other (e.g. network-related or political) reasons to use other VM resources, and the rest of the tutorial takes that into account.

Reply:
The makefile just has some 'regions' coded into is (eu, oz) because the host groups are named like this. This can easily be avoided, I can provide a second "example" hosts file that is much less complicated, which will work for this non-EU gat option